### PR TITLE
Generate: All logits processors are documented and have examples

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

This PR retouches the docstrings of the logits processors such that:
a) they all have a short description of what they do
b) there is a clear example for each class, ideally showcasing its usefulness (note: this will also means the class is doctested)

Moreover, to ensure this information is found, cross references to these classes were added.

Related issue #24575

### CODE REVIEW

File: src/transformers/generation/logits_process.py [hunk 1]
```diff
@@ -100,13 +100,39 @@ def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor, **kwa
...
+
+    Examples:
+
+    ```python
+    >>> from transformers import AutoModelForCausalLM, AutoTokenizer
+
+    >>> tokenizer = AutoTokenizer.from_pretrained("bigscience/bloomz-560m")
+    >>> model = AutoModelForCausalLM.from_pretrained("bigscience/bloomz-560m")
+
+    >>> inputs = tokenizer(["A number:"], return_tensors="pt")
```
Comment (ArthurZucker) [inline src/transformers/generation/logits_process.py:120]:
```suggestion
    >>> inputs = tokenizer("A number:", return_tensors="pt")
```

File: src/transformers/generation/logits_process.py [hunk 2]
```diff
@@ -1294,6 +1466,9 @@ class InfNanRemoveLogitsProcessor(LogitsProcessor):
     r"""
     [`LogitsProcessor`] that removes all `nan` and `inf` values to avoid the generation method to fail. Note that using
     the logits processor should only be used if necessary since it can slow down the generation method.
+
+    This logits processor has no `generate` example, as there shouldn't a correct combination of flags that warrants
+    its use.
```
Comment (ArthurZucker) [inline src/transformers/generation/logits_process.py:1471]:
```suggestion
    This logits processor has no `generate` example, as there shouldn't be a correct combination of flags that warrants
    its use.
```

**Review by ArthurZucker — APPROVED** [review]
LGTM thanks for the cleanup
